controllers/chat/getSignedChannelList.js

```diff
@@ -22,34 +25,71 @@ const __queryChannelBuilder = async (client, userId, limit, offset) => {
   return channels;
 };
 
+// TODO: Change to this filter after admin system has implemented channel image saving;
+// /**
+//  *
+//  * @param {any} acc
+//  * @param {import('stream-chat').Channel} channel
+//  * @returns
+//  */
+// const __filterAndTransformChannelData = (acc, channel) => {
+//   const newChannel = {...channel.data};
+//   delete newChannel.config;
+//   delete newChannel.own_capabilities;
+
+//   const messageLength = channel.state.messages?.length;
+//   const channelType = newChannel?.channel_type;
+
+//   if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return acc;
+
+//   const members = [];
+//   Object.keys(channel.state.members).forEach((member) => {
+//     members.push(channel?.state?.members[member]);
+//   });
+
+//   acc.push({
+//     ...newChannel,
+//     members,
+//     unreadCount: channel.state.unreadCount,
+//     messages: channel.state.messages
+//   });
+
+//   return acc;
+// };
+
 /**
  *
  * @param {import('stream-chat').Channel} channel
  * @returns
  */
-const __filterAndTransformChannelData = (acc, channel) => {
+const __filterAndTransformChannelData = async (channel) => {
   const newChannel = {...channel.data};
   delete newChannel.config;
   delete newChannel.own_capabilities;
 
   const messageLength = channel.state.messages?.length;
   const channelType = newChannel?.channel_type;
 
-  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return acc;
+  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return false;
+
+  if (channelType === CHANNEL_TYPE.TOPIC) {
+    const topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
+    newChannel.image = topic.icon_path || null;
+    newChannel.channelImage = topic.icon_path || null;
+    newChannel.channel_image = topic.icon_path || null;
+  }
 
   const members = [];
   Object.keys(channel.state.members).forEach((member) => {
     members.push(channel?.state?.members[member]);
   });
 
-  acc.push({
+  return {
     ...newChannel,
     members,
     unreadCount: channel.state.unreadCount,
     messages: channel.state.messages
-  });
-
-  return acc;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `__filterAndTransformChannelData` has been refactored to be asynchronous and now includes a database call to fetch topic images. This is a good change as it allows for more dynamic data fetching. However, there are a few potential issues:

1. There's no error handling around the database call at line 76. If the database call fails for any reason (network issues, database down, etc.), it will cause the entire function to fail. Consider adding a try-catch block around this call.

2. The function `removePrefixTopic` is used without checking if `newChannel.id` exists or not. If `newChannel.id` is undefined, it might lead to unexpected behavior in the `removePrefixTopic` function.

3. The properties `image`, `channelImage`, and `channel_image` are all set to the same value (`topic.icon_path || null`). If these properties serve the same purpose, consider using only one to reduce redundancy and potential confusion.

Here are the suggested changes:

```diff
const __filterAndTransformChannelData = async (channel) => {
   const newChannel = {...channel.data};
   delete newChannel.config;
   delete newChannel.own_capabilities;
 
   const messageLength = channel.state.messages?.length;
   const channelType = newChannel?.channel_type;
 
  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return false;

  if (channelType === CHANNEL_TYPE.TOPIC) {
+     if (!newChannel.id) return false; // Add check for newChannel.id
    let topic;
+     try {
      topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
+     } catch (error) {
+       console.error(`Failed to fetch topic: ${error}`);
+       return false;
+     }
    newChannel.image = topic.icon_path || null;
-     newChannel.channelImage = topic.icon_path || null;
-     newChannel.channel_image = topic.icon_path || null;
  }
 
   const members = [];
   Object.keys(channel.state.members).forEach((member) => {
     members.push(channel?.state?.members[member]);
   });
 
  return {
     ...newChannel,
     members,
     unreadCount: channel.state.unreadCount,
     messages: channel.state.messages
  };
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->